hlx_statics/blocks/link/link.js

```diff
@@ -0,0 +1,19 @@
+/**
+ * decorates the link
+ * @param {Element} block The link block element
+ */
+export default async function decorate(block) {
+    const variant = block?.parentElement?.parentElement?.getAttribute('data-variant');
+    const anchorTag = block.querySelector('a');
+    block.appendChild(anchorTag);
+    const nestedDiv = block.querySelector('div');
+    block.removeChild(nestedDiv);
+    if (anchorTag) {
```

Comment:
can we move lines 11 to 18 to link component (https://github.com/adobe/adobe-io-website/blob/link-spectrum-less/hlx_statics/components/link.js#L1) and then call decorateLink here?

(similar to how tab code block , code, and code block are sharing some code: https://github.com/adobe/adobe-io-website/pull/219#issuecomment-2307819184)

this will make it easier to use Link as a component (i.e. something blocks use under the hood). For example, in aio-theme, the following blocks use Link:
- AnchorLink
- Footer
- GatsbyLink
- GlobalHeader
- Heading
- NextPrev
- NextSteps
- OnThisPage

By having something like `decorateLink` under the components dir, we can reuse code between the various blocks.

Reply:
Okay, @melissag-ensemble, i will do it.